app/services/katello/pulp/repository/puppet.rb

```diff
@@ -19,15 +19,15 @@ def generate_mirror_importer
         end
 
         def generate_distributors
-          puppet_install_dist = Runcible::Models::PuppetInstallDistributor.new(puppet_install_distributor_path, :id => repo.pulp_id, :auto_publish => true)
+          puppet_install_dist = Runcible::Models::PuppetInstallDistributor.new(puppet_install_distributor_path, 'modules', :id => repo.pulp_id, :auto_publish => true)
           puppet_dist = Runcible::Models::PuppetDistributor.new(nil, (root.unprotected || false), true,
                                                                 :id => "#{repo.pulp_id}_puppet", :auto_publish => true)
           [puppet_dist, puppet_install_dist]
         end
 
         def puppet_install_distributor_path
           puppet_env = ::Environment.construct_name(repo.organization, repo.environment, repo.content_view)
-          ::File.join(smart_proxy.puppet_path, puppet_env, 'modules')
+          ::File.join(smart_proxy.puppet_path, puppet_env)
```

Comment:
yeah, I think this is a good idea. this PR wants to get rid of the ugly `find --delete` cron we ship, and if we don't upgrade the old distributors it will still be needed